### PR TITLE
Modest change to filtering of coding mutations include ExonicSpliceSite 

### PR DESCRIPTION
**Pull request by iskandr**
Aside from change to use `effect.modifies_protein_sequence` and updating versions of Ensembl used by tests, the rest of the changes are littered with a bunch of PEP8 formatting fixes.

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/varcode/137)

### CODE REVIEW

File: varcode/util.py [hunk 1]
```diff
@@ -48,45 +48,52 @@ def random_variants(
 
     variants = []
 
-    while len(variants) < count:
-        transcript_id = rng.choice(transcript_ids)
-        transcript = ensembl.transcript_by_id(transcript_id)
+    # we should finish way before this loop is over but just in case
+    # something is wrong with PyEnsembl we want to avoid an infinite loop
```
Comment (tavinathanson) [inline varcode/util.py]: What sort of infinite loop are we avoiding here?

File: varcode/effect_ordering.py [hunk 2]
```diff
@@ -12,13 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from .effects import *
+from .effects import (
```
Comment (tavinathanson) [inline varcode/effect_ordering.py]: Why the change?

File: varcode/effect_collection.py [hunk 3]
```diff
@@ -143,8 +143,7 @@ def drop_silent_and_noncoding(self):
         """
         Create a new EffectCollection containing only non-silent coding effects
         """
-        return self.filter(
-            lambda effect: isinstance(effect, NonsilentCodingMutation))
+        return self.filter(lambda effect: effect.modifies_protein_sequence)
```
Comment (tavinathanson) [inline varcode/effect_collection.py]: I think this makes sense?

File: varcode/util.py (hunk 1 above)
Comment (iskandr) [inline varcode/util.py]: The one that occurred to me when Ensembl 83 had bad identifier names for transcripts so none of Transcript objects had valid `sequence` fields.

File: varcode/effect_ordering.py (hunk 2 above)
Comment (iskandr) [inline varcode/effect_ordering.py]: I'm always jumpy about `import *` but no new reason.

File: varcode/effect_collection.py (hunk 3 above)
Comment (iskandr) [inline varcode/effect_collection.py]: That property defaults to False on the base class `MutationEffect` and gets overridden as True on `NonsilentCodingMutation`. The main change in this PR is adding a dynamic property to `ExonicSpliceSite` which checks its alternate effect  -- before that change `effect.modifies_protein_sequence` and `isinstance(effect, NonsilentCodingMutation)` meant the same thing.

File: varcode/util.py [hunk 4]
```diff
@@ -48,45 +48,52 @@ def random_variants(
...
+            transcript_offset = transcript.spliced_offset(base1_genomic_position)
 
-        try:
-            seq = transcript.sequence
-        except ValueError as e:
-            logging.warn(e)
-            # can't get sequence for non-coding transcripts
-            continue
+            try:
+                seq = transcript.sequence
```
Comment (tavinathanson) [inline varcode/util.py]: As discussed offline, this shouldn't catch since it doesn't throw anymore

File: varcode/effect_ordering.py (hunk 2 above)
Comment (tavinathanson) [inline varcode/effect_ordering.py]: :+1:

File: varcode/effect_collection.py (hunk 3 above)
Comment (tavinathanson) [inline varcode/effect_collection.py]: Yeah, that was my understanding of the changes as well; just wondering if there are classes that need to be modified, etc.

File: varcode/util.py (hunk 1 above)
Comment (tavinathanson) [inline varcode/util.py]: :+1:

**Comment by tavinathanson** [comment]
LGTM

**Comment by iskandr** [comment]
Fixes https://github.com/hammerlab/varcode/issues/136

**Comment by arahuja** [comment]
@iskandr do you want to add `modifies_coding_sequence` to `ExonicSpliceSite` as well, also implemented through the alternate effect?

**Comment by iskandr** [comment]
@arahuja Good idea!

**Comment by arahuja** [comment]
@iskandr cool, LGTM
